### PR TITLE
Fix unhandled exception when RuntimeList.xml contains duplicates.

### PR DESCRIPTION
**Pull request by peterhuene**
This PR fixes an unhandled exception that results when RuntimeList.xml
contains entries that have the same destination subpath when resolving the
runtime pack assets.  An error is now emitted instead of the unhandled
exception.

Additionally, this fixes an unhandled exception that occurs when the
RuntimeList.xml file is not present in the runtime pack that is caused by
the exception message not having a NETSDK prefix.

Also removed an unnecessary override of the `Execute` method for
`GenerateDepsFile` task.  This override should have been removed with a
previous cleanup PR, but was overlooked.

Fixes dotnet/cli#11705.

### CODE REVIEW

File: src/Tasks/Common/Resources/Strings.resx
Comment (peterhuene) [inline src/Tasks/Common/Resources/Strings.resx]: I'm definitely open to suggestions regarding these two error messages.

Comment (dsplaisted) [reply src/Tasks/Common/Resources/Strings.resx]: I think there was some discussion in the past about whether we wanted to have links to our repos for filing issues in our error messages. @nguerrera

**Review by dsplaisted — APPROVED** [review]

File: src/Tasks/Common/Resources/Strings.resx
Comment (peterhuene) [reply src/Tasks/Common/Resources/Strings.resx]: This is primarily why I wanted suggestions 😄.  These are internal errors that are not actionable by users directly and I'd love to point them towards a way to inform us of the problem, even if it's really unlikely users would ever run into these before we do.  But I can understand not wanting repo references in the error messages.

Comment (nguerrera) [reply src/Tasks/Common/Resources/Strings.resx]: We could make an aka.ms/report-runtime-list-issue or something?

I don't actually recall such a discussion at this point. cc @livarcocc

Comment (peterhuene) [reply src/Tasks/Common/Resources/Strings.resx]: Maybe a more generic `aka.ms/dotnet-core-issue` that takes them to `dotnet/core` and we can triage from there?

Comment (nguerrera) [reply src/Tasks/Common/Resources/Strings.resx]: That works for me.

Comment (peterhuene) [reply src/Tasks/Common/Resources/Strings.resx]: What about using `aka.ms/dotnet-core-sdk-issue` and we can point it at either the CLI or SDK repos? It's likely that the issue really belongs to core-setup for these two errors, but I think we should be responsible for triage given the errors are coming from the SDK.

We can then use the aka.ms link for any other internal SDK errors in the future.

Comment (nguerrera) [reply src/Tasks/Common/Resources/Strings.resx]: Ship it.

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs
Comment (nguerrera) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs]: Should this be OrdinalIgnoreCase?

**Review by nguerrera — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs
Comment (peterhuene) [reply src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs]: It should be; even if the file system is case-sensitive, we wouldn't ever want case-insensitive duplicates in the list.

I'll fix.

**Comment by peterhuene** [comment]
Waiting on some feedback from @KathleenDollard before merging.

**Pull request merged**
